### PR TITLE
Claim get all open and update status

### LINKED ISSUE #67: getAllClaims
- Endpoint URI: */claims*
- Http Method: *GET*
- API Logic: 
    - get all claims in OPEN Status
    - Sort based on event, createdDate
- Authority
    - CC Head
    - Administrator
- Progress
    - [x] development
    - [ ] integration Test

### LINKED ISSUE #68: updateClaimStatus
- Endpoint URI: */claims/{claimId}/status/{claimStatus}*
- Http Method: *PUT*
- API Logic: 
    - check claimId exists
    - Only claim status is OPEN can be update status
    - update claimStatus
- Authority
    - CC Head
    - Administrator
- Progress
    - [x] development
    - [ ] integration Test

### PR DESCRIPTION
**Pull request by CarlosSia**
Resolve #67 
Resolve #68

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java [hunk 1]
```diff
@@ -46,84 +49,38 @@ public ClaimResource(ClaimService claimService) {
...
      */
-    @PutMapping("/claims")
-    public ResponseEntity<ClaimDTO> updateClaim(@RequestBody ClaimDTO claimDTO) throws URISyntaxException {
-        log.debug("REST request to update Claim : {}", claimDTO);
-        if (claimDTO.getId() == null) {
-            throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "idnull");
-        }
-        ClaimDTO result = claimService.save(claimDTO);
+    @PutMapping("/claims/{id}/status/{claimStatus}")
+    @PreAuthorize("hasRole(\"" + AuthoritiesConstants.ADMIN + "\") || @managementTeamSecurityExpression.isCurrentAdministrator()")
```
Comment (kachun333) [inline src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java]: please don't use `hasRole(\"" + AuthoritiesConstants.ADMIN + "\")` anymore
use `managementTeamSecurityExpression` instead

File: src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java [hunk 2]
```diff
@@ -46,84 +49,38 @@ public ClaimResource(ClaimService claimService) {
...
+     * {@code GET  /claims} : get all the claims with Open ClaimStatus.
      *
      * @param pageable the pagination information.
      * @param queryParams a {@link MultiValueMap} query parameters.
      * @param uriBuilder a {@link UriComponentsBuilder} URI builder.
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of claims in body.
      */
     @GetMapping("/claims")
-    public ResponseEntity<List<ClaimDTO>> getAllClaims(Pageable pageable, @RequestParam MultiValueMap<String, String> queryParams, UriComponentsBuilder uriBuilder) {
+    @PreAuthorize("hasRole(\"" + AuthoritiesConstants.ADMIN + "\") || @managementTeamSecurityExpression.isCurrentAdministrator()")
```
Comment (kachun333) [inline src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java]: same issue. don't use `"hasRole(\"" + AuthoritiesConstants.ADMIN + "\")`

**Review by kachun333 — DISMISSED** [review]

File: src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java (hunk 1 above)
Comment (CarlosSia) [reply src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java]: Resolve by https://github.com/Five-Fishes/Club-Management-Spring-Server/pull/98/commits/fe3e1946a1a7d5606072eb3fc2df3efbff6a8015

File: src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java (hunk 2 above)
Comment (CarlosSia) [reply src/main/java/com/thirdcc/webapp/web/rest/ClaimResource.java]: Resolved by https://github.com/Five-Fishes/Club-Management-Spring-Server/pull/98/commits/fe3e1946a1a7d5606072eb3fc2df3efbff6a8015

**Review by LUXIANZE — APPROVED** [review]

**Review by HowardYaw — APPROVED** [review]

**Pull request merged**